paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/lookup/FileStoreLookupFunction.java

```diff
@@ -120,10 +139,20 @@ public FileStoreLookupFunction(
     }
 
     public void open(FunctionContext context) throws Exception {
-        String tmpDirectory = getTmpDirectory(context);
+        Options options = Options.fromMap(table.options());
+
+        this.lookupAsync = options.get(FlinkConnectorOptions.LOOKUP_ASYNC);
+        if (lookupAsync) {
+            this.lock = new ReentrantLock();
+            this.asyncThreadPool =
+                    Executors.newFixedThreadPool(
+                            ASYNC_THREAD_POOL_SIZE,
```

Comment:
Actually, `FileStoreLookupFunction` is not a thread safe class.
You can take a look to `RocksDBState.serializeKey`.

So the thread number should alway be 1 only.

Reply:
Got it, thanks for the reminder.